java/src/hdf/hdf5lib/H5.java

```diff
@@ -1139,760 +1211,2029 @@ else if (is1D && (dataClass.getComponentType() == String.class)) {
+     * @param isCriticalPinning
+     *            IN: request lock on data reference.
+     *
+     * @return a non-negative value if successful
+     *
+     * @exception HDF5LibraryException
+     *                - Error from the HDF-5 Library.
+     * @exception NullPointerException
+     *                - data is null.
+     **/
+    public synchronized static native int H5Awrite_short(long attr_id, long mem_type_id, short[] buf, boolean isCriticalPinning)
+            throws HDF5LibraryException, NullPointerException;
+
+    /**
+     * H5Awrite writes an attribute, specified with attr_id. The attribute's memory datatype is specified with
+     * mem_type_id. The entire attribute is written from buffer of short to the file.
+     *
+     * @param attr_id
+     *            IN: Identifier of an attribute to write.
+     * @param mem_type_id
+     *            IN: Identifier of the attribute datatype (in memory).
+     * @param buf
+     *            IN: Buffer of short with data to be written to the file.
+     *
+     * @return a non-negative value if successful
+     *
+     * @exception HDF5LibraryException
+     *                - Error from the HDF-5 Library.
+     * @exception NullPointerException
+     *                - data is null.
+     **/
+    public synchronized static int H5Awrite_short(long attr_id, long mem_type_id, short[] buf)
+            throws HDF5LibraryException, NullPointerException
+    {
+        return H5Awrite_short(attr_id, mem_type_id, buf, true);
+    }
+
+    /**
+     * H5Awrite writes an attribute, specified with attr_id. The attribute's memory datatype is specified with
+     * mem_type_id. The entire attribute is written from buffer of string to the file.
+     *
+     * @param attr_id
+     *            IN: Identifier of an attribute to write.
+     * @param mem_type_id
+     *            IN: Identifier of the attribute datatype (in memory).
+     * @param buf
+     *            IN: Buffer of string with data to be written to the file.
+     *
+     * @return a non-negative value if successful
+     *
+     * @exception HDF5LibraryException
+     *                - Error from the HDF-5 Library.
+     * @exception NullPointerException
+     *                - data is null.
+     **/
+    public synchronized static native int H5Awrite_string(long attr_id, long mem_type_id, String[] buf)
+            throws HDF5LibraryException, NullPointerException;
+
+    /**
+     * H5Awrite writes an attribute, specified with attr_id. The attribute's memory datatype is specified with
+     * mem_type_id. The entire attribute is written from buffer of variable-lenght to the file.
+     *
+     * @param attr_id
+     *            IN: Identifier of an attribute to write.
+     * @param mem_type_id
+     *            IN: Identifier of the attribute datatype (in memory).
+     * @param buf
+     *            IN: Buffer of variable-lenght with data to be written to the file.
```

Comment:
typo - variable-length.  Repeated in 9 lines in H5.java (also in develop).

Reply:
That is in all branches - will need to be fixed in the future.